uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/WorkPatternCalendar.php

```diff
@@ -36,8 +36,19 @@ class CRM_HRLeaveAndAbsences_Service_WorkPatternCalendar {
    */
   private $defaultWorkPattern;
 
-  public function __construct($contactID, AbsencePeriod $absencePeriod, JobContractService $jobContractService) {
+  /**
+   * CRM_HRLeaveAndAbsences_Service_WorkPatternCalendar constructor.
+   *
+   * @param int $contactID
+   * @param string $startDate
+   * @param string $endDate
+   * @param \CRM_HRLeaveAndAbsences_Service_JobContract $jobContractService
+   */
+  public function __construct($contactID, $startDate, $endDate, JobContractService $jobContractService) {
     $this->contactID = $contactID;
+    $absencePeriod = new AbsencePeriod();
+    $absencePeriod->start_date = CRM_Utils_Date::processDate($startDate, null, false, 'Y-m-d');
+    $absencePeriod->end_date = CRM_Utils_Date::processDate($endDate, null, false, 'Y-m-d');
```

Comment:
Actually what is needed is just an object that has the start_date and end_date property, even an stdClass should suffice. Not sure if it will be worth modifying the `CRM_HRCore_Date_BasicDatePeriod` class as it also does does things already in the constructor which we don't need, like modifying the end date.
Also another reason I opted for the current approach is because the Absence Period BAO function `adjustDatesToMatchPeriodDates` is used in the class.